### PR TITLE
Typings

### PR DESCRIPTION
**Pull request by apoco**
Add TypeScript declarations to aid developers

### CODE REVIEW

**Comment by apoco** [comment]
Thanks for sharing this code; it was just what I was looking for. I thought I'd share my TypeScript declarations for the module with you in case you want to publish a new version that will include prompting for TS-enabled development environments.

**Comment by felixfbecker** [comment]
@philfreo is there any chance we could get this released? It would be super helpful 🙇🏻

File: src/addresscompiler.d.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
+declare module "addresscompiler" {
+  type Item = Address | Group;
```
Comment (felixfbecker) [inline src/addresscompiler.d.ts]:
```suggestion
  export type Item = Address | Group;
```

File: src/addresscompiler.d.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+  type Address = {
```
Comment (felixfbecker) [inline src/addresscompiler.d.ts]:
```suggestion
  export interface Address {
```

File: src/addresscompiler.d.ts [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+    name?: string | null;
+    address: string;
+  };
+
+  type Group = {
```
Comment (felixfbecker) [inline src/addresscompiler.d.ts]:
```suggestion
  export interface Group {
```

**Review by felixfbecker — COMMENTED** [review]
Made some suggestions to export the types so they can be referenced in code

**Comment by apoco** [comment]
@felixfbecker I've committed those changes, but I'm not sure that a module declaration supports having both named exports and `export = `-style CommonJS exports. Seems to be working in my project, though.

**Comment by felixfbecker** [comment]
As long as it's only types it's okay, it's the only way to export types in addition to a single `export =` value 🙂

**Comment by marcotisi** [comment]
Thanks for your contributions, the package has been released under `v1.1.0` 🎉

**Pull request merged**